src/box/lua/experimental/connpool.lua

```diff
@@ -0,0 +1,60 @@
+local config = require('config')
+local checks = require('checks')
+local netbox = require('net.box')
+
+local connections = {}
+
+local function is_connection_valid(conn, opts)
+    if conn == nil or conn.state == 'error' or conn.state == 'closed' then
+        return false
+    end
+    assert(type(opts) == 'table')
+    local conn_opts = conn.opts or {}
+    if opts.fetch_schema ~= false and conn_opts.fetch_schema == false then
+        return false
+    end
+    return true
+end
+
+local function connect(instance_name, opts)
+    checks('string', {
+        connect_timeout = '?number',
+        wait_connected = '?boolean',
+        fetch_schema = '?boolean',
+    })
+    opts = opts or {}
+
+    local conn = connections[instance_name]
```

Comment:
Yet another thing that I would solve in a separate PR.

It seems, if instance's URI is changed, we don't re-create a connection.

Reply:
I created a new issue about this: https://github.com/tarantool/tarantool/issues/9910